tests/unit/express-checkout/test-class-wc-payments-express-checkout-button-helper.php

```diff
@@ -208,6 +209,30 @@ function () {
 		remove_all_filters( 'wcpay_payment_request_total_label_suffix' );
 	}
 
+	public function test_should_show_express_checkout_button_for_tokenized_ece_with_billing_email() {
+		global $wp;
+		global $wp_query;
+
+		$this->mock_wcpay_account
+			->method( 'is_stripe_connected' )
+			->willReturn( true );
+		WC_Payments::mode()->dev();
+		$_GET['pay_for_order'] = true;
+
+		// Total is 100 USD, which is above both payment methods (Affirm and AfterPay) minimums.
+		$order                = WC_Helper_Order::create_order( 1, 100 );
+		$order_id             = $order->get_id();
+		$wp->query_vars       = [ 'order-pay' => strval( $order_id ) ];
+		$wp_query->query_vars = [ 'order-pay' => strval( $order_id ) ];
+
+		update_option( '_wcpay_feature_tokenized_cart_ece', '1' );
+		add_filter( 'woocommerce_is_checkout', '__return_true' );
+
+		$this->assertTrue( $this->system_under_test->should_show_express_checkout_button() );
+
+		remove_filter( 'woocommerce_is_checkout', '__return_true' );
+	}
```

Comment:
I attempted to create a test to negate the situation (e.g.: create a test to verify the button isn't rendered)

However, unsetting the billing email via either `$order->set_billing_address( [ 'email' => '' ] )` or `$order->set_billing_email('')` didn't work.
I thought of querying the DB directly, but with HPOS and non-HPOS alternatives, I didn't think it was maintainable.

So here we are 🤷

Reply:
I tried for a bit on this as well. The email really does not want to update. I'm getting strange results with the `$order->save()` method. The same order ID is returned every time and no logs set within that method are coming through. So I'm with you on this one. Directly modifying the DB seems to be the only way to remove the email address.